tests/xharness/Microsoft.DotNet.XHarness.iOS.Shared/Listeners/SimpleListener.cs

```diff
@@ -32,7 +32,7 @@ public abstract class SimpleListener : ISimpleListener, IDisposable {
 		protected abstract void Start ();
 		protected abstract void Stop ();
 
-		public Task ConnectedTask => connected.Task;
+		public Task<bool> ConnectedTask => connected.Task;
```

Comment:
I don't think I quite understand why this fixes anything.

The `ConnectedTask` property is used in one place:

https://github.com/xamarin/xamarin-macios/blob/764734d20aa0862ff0666559615762ec709bcf07/tests/xharness/AppRunner.cs#L293-L296

and it doesn't look like it should matter whether `ConnectedTask` returns a `Task<bool>` or a `Task`, the bool return value isn't used.

Reply:
It is used, in the callback as the result of the launch, that is precisely the issue, it uses Task.Result:

https://github.com/xamarin/xamarin-macios/blob/main/tests/xharness/Microsoft.DotNet.XHarness.iOS.Shared/TestReporter.cs#L219

But we can see the device tests since they'll be triggered.

Reply:
`testReporter.LaunchCallback` takes the result from `TimeoutAfter`, which is a new `Task`, not the one returned from `ConnectedTask`:

https://github.com/xamarin/xamarin-macios/blob/764734d20aa0862ff0666559615762ec709bcf07/tests/xharness/Microsoft.DotNet.XHarness.iOS.Shared/Utilities/Extensions.cs#L85-L95

thus it shouldn't matter if `ConnectedTask` returns a `Task<bool>` or `Task`, because that task doesn't get to `LaunchCallback` in any case.

Reply:
@rolfbjarne we do have successful results: http://xamarin-storage/jenkins/xamarin-macios/fix-xharness-device/6ad071cfa5d9407249744ec46d6d75c6b4f7908f/4110735/device-tests/jenkins-results/tests/index.html

Reply:
@mandel-macaque that looks similar to http://xamarin-storage/jenkins/xamarin-macios/main/e343e68725000041a519e243fe6f84edcbad1916/4111016/device-tests/jenkins-results/tests/index.html, which doesn't have the fix